cmd/helm/install.go

```diff
@@ -160,7 +163,8 @@ func newInstallCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 
 func addInstallFlags(cmd *cobra.Command, f *pflag.FlagSet, client *action.Install, valueOpts *values.Options) {
 	f.BoolVar(&client.CreateNamespace, "create-namespace", false, "create the release namespace if not present")
-	f.BoolVar(&client.DryRun, "dry-run", false, "simulate an install")
+	f.StringVar(&client.DryRunOption, "dry-run", "unchanged", "simulate an install. If --dry-run is set with no option being specified or as '--dry-run=client', it will not attempt cluster connections. Setting '--dry-run=server' allows attempting cluster connections.")
+	f.Lookup("dry-run").NoOptDefVal = "client"
```

Comment:
Great call out. I'll change it to `""` so end users are not exposed to this internal default value.

Reply:
@tapaskapadia I think this is my one nit I would like to see changed before approving.

Reply:
@mattfarina I just pushed up the changes to make the default value to be "" which make the help documentation much clearer. Here's how the help documentation reads now:
```
--dry-run string[="client"]                  simulate an install. If --dry-run is set with no option being specified or as '--dry-run=client', it will not attempt cluster connections. Setting '--dry-run=server' allows attempting cluster connections.

```
Here's the expected behavior table (nothing has changed) with my matching test results including the latest changes: [test_03-30-23.txt](https://github.com/helm/helm/files/11363626/test_03-30-23.txt)

| command       | dryRun Flag Value | expected behavior |
| ------------- | ------------- | ------------- |
| install  | --dry-run  | dry run w/ no server interaction  |
| install  | --dry-run=true | dry run w/ no server interaction  |
| install  | --dry-run=client | dry run w/ no server interaction  |
| install  | --dry-run=server | dry run with server interaction  |
| install  | --dry-run=badinput  | error for invalid option |
| install  | --dry-run=none  | not a dry run  |
| install  | --dry-run=false  | not a dry run  |
| install  | [no flag present]  | not a dry run  |
| upgrade --install  | --dry-run  | dry run w/ no server interaction  |
| upgrade --install  | --dry-run=true | dry run w/ no server interaction  |
| upgrade --install  | --dry-run=client | dry run w/ no server interaction  |
| upgrade --install  | --dry-run=server | dry run with server interaction  |
| upgrade --install  | --dry-run=badinput | error for invalid option  |
| upgrade --install  | --dry-run=none  | not a dry run  |
| upgrade --install  | --dry-run=false  | not a dry run  |
| upgrade --install  | [no flag present]  | not a dry run  |
| template  | --dry-run | dry run w/ no server interaction  |
| template  | --dry-run=client | dry run w/ no server interaction  |
| template  | --dry-run=server | dry run with server interaction  |
| template  | --dry-run=true | dry run w/ no server interaction  |
| template  | --dry-run=false | dry run with server interaction  |
| template  | --dry-run=badinput | error for invalid option |
| template  | --dry-run=none | dry run with server interaction  |
| template  | [no flag present] | dry run w/ no server interaction  |

Reply:
Is there any further action needed here? I'm happy to help in any way possible.

Reply:
@mattfarina @tapaskapadia Are there any additional changes required before this can be merged? It seems like all of the requested changes have been addressed.

Reply:
@philomory I believe I have implemented all the requested changes. If there’s anything else needed, I happy to make changes.